native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Encrypt.java

```diff
@@ -80,4 +91,23 @@ public static Object encryptRsaEcb(BArray inputValue, Object keys, Object paddin
         return CryptoUtils.rsaEncryptDecrypt(CryptoUtils.CipherMode.ENCRYPT, Constants.ECB, padding.toString(), key,
                 input, null, -1);
     }
+
+    public static Object encryptPgp(BArray plainTextValue, BArray publicKeyValue, BMap options) {
+        byte[] plainText = plainTextValue.getBytes();
+        byte[] publicKey = publicKeyValue.getBytes();
+        InputStream publicKeyStream = new ByteArrayInputStream(publicKey);
```

Comment:
When we create `InputStream` the best practice is to close them once the required work is done. And I think it is better to do that in the same place where we create the stream using a `try-with-resources` block.

Reply:
I have addressed this in https://github.com/ballerina-platform/module-ballerina-crypto/pull/555/commits/78e3fb4fbeab147e5c59d3a51cfec372b405651c